MapboxCoreNavigation/CLLocation.swift

```diff
@@ -71,14 +71,16 @@ extension CLLocation {
     
     //MARK: - Route Snapping
     
-    func snapped(to legProgress: RouteLegProgress) -> CLLocation? {
+    func snapped(to legProgress: RouteLegProgress, previousSpeeds: [CLLocationSpeed]) -> CLLocation? {
```

Comment:
This makes route snapping stateful, no longer purely a function of a single CLLocation. So this method should be moved out of the CLLocation extension and into RouteController, which stores the previous speeds as an ivar.